
Based on https://git.parabola.nu/abslibre.git/diff/libre/iceweasel/9004-FSDG-misc.patch?id=0947239827274ca70e91a8da93219846bdad401a
- Remove 'More from Mozilla' section in Preferences
- Remove promos of Firefox Focus and Klar from Private Browsing

diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index cc42846d..52d35d49 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -1388,11 +1388,11 @@ pref("accessibility.typeaheadfind.linksonly", false);
 pref("accessibility.typeaheadfind.flashBar", 1);
 
 // Whether we had to hide the "Firefox Labs" section because it would be empty.
-pref("browser.preferences.experimental.hidden", false);
+pref("browser.preferences.experimental.hidden", true);
 // Whether we show the "More from Mozilla" section.
-pref("browser.preferences.moreFromMozilla", true);
+pref("browser.preferences.moreFromMozilla", false);
 // Whether we show the "AI Controls" pane.
-pref("browser.preferences.aiControls", true);
+pref("browser.preferences.aiControls", false);
 // Whether to show unavailable AI controls regardless of region/locale
 // restrictions. This is intended for localizers to be able to see the strings.
 pref("browser.preferences.aiControls.showUnavailable", false);
@@ -2685,7 +2685,7 @@ pref("browser.vpn_promo.enabled", false);
 pref("browser.contentblocking.report.vpn_regions", "as,at,au,bd,be,bg,br,ca,ch,cl,co,cy,cz,de,dk,ee,eg,es,fi,fr,gb,gg,gr,hr,hu,id,ie,im,in,io,it,je,ke,kr,lt,lu,lv,ma,mp,mt,mx,my,ng,nl,no,nz,pl,pr,pt,ro,sa,se,sg,si,sk,sn,th,tr,tw,ua,ug,uk,um,us,vg,vi,vn,za");
 
 // Default to enabling pin promos to be shown where allowed.
-pref("browser.promo.pin.enabled", true);
+pref("browser.promo.pin.enabled", false);
 
 pref("browser.contentblocking.report.hide_vpn_banner", false);
 pref("browser.contentblocking.report.vpn_sub_id", "sub_HrfCZF7VPHzZkA");
diff -Nru a/browser/base/content/test/static/browser_all_files_referenced.js b/browser/base/content/test/static/browser_all_files_referenced.js
--- a/browser/base/content/test/static/browser_all_files_referenced.js	2022-05-22 09:35:55.336673325 +0000
+++ b/browser/base/content/test/static/browser_all_files_referenced.js	2022-05-22 21:50:33.783348501 +0000
@@ -188,12 +188,6 @@ var allowlist = [
   // extensions/pref/autoconfig/src/nsReadConfig.cpp
   { file: "resource://gre/defaults/autoconfig/prefcalls.js" },
 
-  // browser/components/preferences/moreFromMozilla.js
-  // These files URLs are constructed programatically at run time.
-  {
-    file: "chrome://browser/content/preferences/more-from-mozilla-qr-code-simple.svg",
-  },
-
   { file: "resource://gre/greprefs.js" },
 
   // toolkit/mozapps/extensions/AddonContentPolicy.cpp
diff --git a/browser/components/asrouter/modules/PanelTestProvider.sys.mjs b/browser/components/newtab/lib/PanelTestProvider.sys.mjs
index b5ff592..22cdf39 100644
--- a/browser/components/asrouter/modules/PanelTestProvider.sys.mjs	2022-10-16 05:35:50.835203327 -0500
+++ b/browser/components/asrouter/modules/PanelTestProvider.sys.mjs	2022-10-16 06:24:48.095827505 -0500
@@ -367,85 +374,6 @@ const MESSAGES = () => [
     frequency: { lifetime: 3 },
     trigger: { id: "defaultBrowserCheck" },
   },
-  {
-    id: "PB_FOCUS_PROMO",
-    groups: ["panel-test-provider"],
-    template: "spotlight",
-    content: {
-      template: "multistage",
-      backdrop: "transparent",
-      screens: [
-        {
-          id: "PBM_FIREFOX_FOCUS",
-          order: 0,
-          content: {
-            logo: {
-              imageURL: "chrome://browser/content/assets/focus-logo.svg",
-              height: "48px",
-            },
-            title: {
-              string_id: "spotlight-focus-promo-title",
-            },
-            subtitle: {
-              string_id: "spotlight-focus-promo-subtitle",
-            },
-            dismiss_button: {
-              action: {
-                dismiss: true,
-              },
-            },
-            ios: {
-              action: {
-                data: {
-                  args: "https://app.adjust.com/167k4ih?campaign=firefox-desktop&adgroup=pb&creative=focus-omc172&redirect=https%3A%2F%2Fapps.apple.com%2Fus%2Fapp%2Ffirefox-focus-privacy-browser%2Fid1055677337",
-                  where: "tabshifted",
-                },
-                type: "OPEN_URL",
-                navigate: true,
-              },
-            },
-            android: {
-              action: {
-                data: {
-                  args: "https://app.adjust.com/167k4ih?campaign=firefox-desktop&adgroup=pb&creative=focus-omc172&redirect=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dorg.mozilla.focus",
-                  where: "tabshifted",
-                },
-                type: "OPEN_URL",
-                navigate: true,
-              },
-            },
-            email_link: {
-              action: {
-                data: {
-                  args: "https://mozilla.org",
-                  where: "tabshifted",
-                },
-                type: "OPEN_URL",
-                navigate: true,
-              },
-            },
-            tiles: {
-              type: "mobile_downloads",
-              data: {
-                QR_code: {
-                  image_url:
-                    "chrome://browser/content/assets/focus-qr-code.svg",
-                  alt_text: {
-                    string_id: "spotlight-focus-promo-qr-code",
-                  },
-                },
-                email: {
-                  link_text: "Email yourself a link",
-                },
-                marketplace_buttons: ["ios", "android"],
-              },
-            },
-          },
-        },
-      ],
-    },
-    trigger: { id: "defaultBrowserCheck" },
-  },
   {
     id: "PB_NEWTAB_VPN_PROMO",
     template: "pb_newtab",
diff --git a/browser/components/preferences/jar.mn b/browser/components/preferences/jar.mn
index de83282c..3dec49ff 100644
--- a/browser/components/preferences/jar.mn
+++ b/browser/components/preferences/jar.mn
@@ -17,7 +17,6 @@ browser.jar:
    content/browser/preferences/fxaPairDevice.xhtml
    content/browser/preferences/fxaPairDevice.js
    content/browser/preferences/findInPage.js
-   content/browser/preferences/more-from-mozilla-qr-code-simple.svg
    content/browser/preferences/browser-layout-horizontal.svg
    content/browser/preferences/browser-layout-vertical.svg
    content/browser/preferences/web-appearance-dark.svg
diff --git a/browser/components/preferences/moreFromMozilla.js b/browser/components/preferences/moreFromMozilla.js
index 044157bc..89c9640b 100644
--- a/browser/components/preferences/moreFromMozilla.js
+++ b/browser/components/preferences/moreFromMozilla.js
@@ -81,46 +81,7 @@ var gMoreFromMozillaPane = {
 
   renderProducts() {
     const isRegionUS = Region.home.toLowerCase() === "us";
-    let products = [
-      {
-        id: "firefox-mobile",
-        title_string_id: "more-from-moz-firefox-mobile-title",
-        description_string_id: "more-from-moz-firefox-mobile-description",
-        region: "global",
-        button: {
-          id: "fxMobile",
-          type: "link",
-          label_string_id: "more-from-moz-learn-more-link",
-          actionURL: "https://www.mozilla.org/firefox/browsers/mobile/",
-        },
-        qrcode: {
-          title: {
-            string_id: "more-from-moz-qr-code-box-firefox-mobile-title",
-          },
-          image_src_prefix:
-            "chrome://browser/content/preferences/more-from-mozilla-qr-code",
-          button: {
-            id: "qr-code-send-email",
-            label: {
-              string_id: "more-from-moz-qr-code-box-firefox-mobile-button",
-            },
-            actionURL: "https://www.mozilla.org/firefox/mobile/get-app/?v=mfm",
-          },
-        },
-      },
-      {
-        id: "mozilla-monitor",
-        title_string_id: "more-from-moz-mozilla-monitor-title",
-        description_string_id:
-          "more-from-moz-mozilla-monitor-global-description",
-        region: isRegionUS ? "us" : "global",
-        button: {
-          id: "mozillaMonitor",
-          label_string_id: "more-from-moz-mozilla-monitor-button",
-          actionURL: "https://monitor.mozilla.org/",
-        },
-      },
-    ];
+    let products = [];
 
     if (BrowserUtils.shouldShowVPNPromo()) {
       const vpn = {
